src/Pool.sol

```diff
@@ -419,14 +445,21 @@ abstract contract Pool {
             if (receiverAddr == ReceiverWeightsImpl.ADDR_ROOT) break;
             int128 amtPerSecDelta = int128(uint128(weight)) * amtPerWeightPerSecDelta;
             _setReceiverDeltaFromNow(receiverAddr, amtPerSecDelta, timeEnd);
-            if (amtPerSecDelta > 0) {
-                // Sending is starting
-                uint128 amtPerSec = uint128(amtPerSecDelta);
-                emit SenderToReceiverUpdated(id, receiverAddr, amtPerSec, timeEnd);
-            } else {
-                // Sending is stopping
-                emit SenderToReceiverUpdated(id, receiverAddr, 0, _currTimestamp());
+
+            // Find an existing receiver stream log to update
+            uint256 updated = 0;
+            while (updated < updates.length && updates.updates[updated].receiver != receiverAddr) {
+                updated++;
+            }
+            // New receiver, will be pushed
+            if (updated == updates.length) {
+                updates.length++;
             }
+            updates.updates[updated] = StreamUpdate({
+                receiver: receiverAddr,
+                amtPerSec: amtPerSecDelta > 0 ? uint128(amtPerSecDelta) : 0,
+                endTime: amtPerSecDelta > 0 ? timeEnd : _currTimestamp()
+            });
```

Comment:
Just for understanding.

Lets  say I have already a stream starting from  
```
t=0
endTime=5
amtPerSec=1
```

Let say I now increase my topUp (enough for t=10 to end and my amtPerSec at  block.timestamp=3.

Currently in the stopSending we would throw an event with:
SenderToReceiverUpdated  amPerSec=1, t=3 (in stopSending)
SenderToReceiverUpdated amPerSec=2 t= 10 (in startSending)

with the new logic we would overwrite the updates array entry and only throw a 
SenderToReceiverUpdated amPerSec=2 t= 10 event.

Correct?

Reply:
Correct. While the current one is technically correct, it's excessive and causes noise, which needs to be handled when processing events.

Reply:
Yeah, we should maybe compare the gas differences if it is worth removing the event noise.